app/services/katello/product_content_importer.rb

```diff
@@ -76,13 +36,12 @@ def add_product_content(product, product_content_json)
     end
 
     def find_product_for_content(content_id)
-      #Hash.find converts hash into enumerable iterating over each key,value pair which we can capture based on find result
-      cp_product_id, __cp_product_value = @cp_products.find do |prod_json|
-        prod_json&.last&.first&.[]('productContent')&.any? do |product_content_json|
+      prod = @cp_products.find do |prod_json|
```

Comment:
```suggestion
      prod = @cp_products.find do |prod_json|
      next unless prod_json&. []('productContent').present?
```

Reply:
May be even 
`prod_json&.[]('productContent')&.any? do |product_content_json|` instead of `prod_json['productContent'].any? do |product_content_json|`